src/udev/udev-builtin-path_id.c

```diff
@@ -200,6 +202,16 @@ static sd_device *handle_scsi_sas(sd_device *parent, char **path) {
                 /* Get expander's address */
                 if (sd_device_get_sysattr_value(expander_sasdev, "sas_address", &sas_address) < 0)
                         return NULL;
+
+                /* Use expander-side phy id */
+                if (sd_device_get_sysname(port, &sysname) < 0)
+                        return NULL;
+                if (!snprintf_ok(phyname, sizeof phyname, "phy-%s", sysname + strlen("port-")))
+                        return NULL;
+                if (sd_device_new_from_subsystem_sysname(&port_sasdev, "sas_phy", phyname) < 0)
+                        return NULL;
+                if (sd_device_get_sysattr_value(port_sasdev, "phy_identifier", &phy_id) < 0)
+                        return NULL;
```

Comment:
If `phy_id` is successfully obtained here, then it seems not necessary to obtain the value from `target_sasdev`.
So, how about voidifying this call, and moving the above lines around `if (sd_device_get_sysattr_value(target_sasdev, "phy_identifier", &phy_id) < 0)` after this block as a fallback logic?

Reply:
Agreed, I'll make that change.